docs/README.md

```diff
@@ -0,0 +1,208 @@
+
+As such, there must be enough information implied or in the message passed to the endpoint to allow the Agency to decrypt the message.  Further, there must be enough information after the decryption for the Agency to be able to determine what Cloud Agent should handle the message further.
+
+#### Cloud Agent Message Handling
+The Cloud Agent must be able to respond to the payload of the message and be able to contact as necessary its associated Edge Agent(s). There are several classes of messages for the Cloud Agent to handle:
+
+* Messages from other Agents to be forwarded to the Edge Agent. Such messages may be have to be queued for the Edge Agent to receive when they are online.
+* Messages from the Edge Agent to the Agent of other Identities. Such messages need to be forwarded to the endpoint of the of the other Identity. The Cloud Agent may hold the necessary information for that action, or receive that with the message from the Edge Agent.
+* Messages from the Edge Agent for administrative purposes. These are used for things like configuring the Cloud Agent, backups, restores and so on.
+
+#### Cloud Agent Keys and Endpoints
+The Cloud and Edge Agents must have keypairs so that two Agents can auth-crypt (encrypt and sign) messages to one another. Based our core definition of the Cloud Agent being primarily for message transport, the Cloud Agent does not have keys for accessing the Application Layer content of messages - only the Edge Agents have those keys.
+
+Mobile Edge Agents will be definition have a non-persistent endpoint and the Cloud Agent must have a way to message the Mobile Agent Agent, including queuing messages when the Edge Agent is offline. <<<How does this happen in the mobile world?  The Edge Agent maintains the connection with the Cloud Agent?>>>
+
+#### Other Cloud Agent Functions
+A Cloud Agent may have other functions beyond message transport. 
+
+A key feature might be providing Wallet backup and restore functionality - holding an encrypted version of an Edge Agent’s wallet. The encryption and encryption key handling would be on the Agent side, but the encrypted data could be stored with the Cloud Agent. The encryption key handling would involve all of the DKMS mechanisms described in the [Evernym/DHS document](https://github.com/hyperledger/indy-sdk/blob/677a0439487a1b7ce64c2e62671ed3e0079cc11f/doc/design/005-dkms/DKMS%20Design%20and%20Architecture%20V3.md). So many details to be added…
+
+#### Variations
+In an Institution environment, a Cloud Agent is less important since the Edge Agent is likely a persistent end point using on-premise or Cloud hosting. It may still be useful to have a Cloud Agent to prevent correlation. Regardless, an IEA without a Cloud Agent should still look to the rest of the world like any other Agent, so it should have endpoint message handling like that of a Cloud Agent - e.g. expect to be sent anon-crypt messages with a transport instructions and an encrypted payload.
```

Comment:
"expect to be sent anon-crypt messages with (*a* omitted) transport instructions and an encrypted payload" or some other wording? It reads a little funny at the moment.